bookmaker_titlepage.rb

```diff
@@ -153,152 +332,92 @@ def getResourceDir(imprint, json)
-allimg = File.join(submitted_images, "*")
-etparr = Dir[allimg].select { |f| f.include?('epubtitlepage.')}
-ptparr = Dir[allimg].select { |f| f.include?('titlepage.')}
-
-newtitlepage = ""
-
-if etparr.any?
-  puts "Found a new EPUB titlepage"
-  newtitlepage = etparr.find { |e| /[\/|\\]epubtitlepage\./ =~ e }
-elsif ptparr.any?
-  puts "Found a new POD titlepage"
-  newtitlepage = ptparr.find { |e| /[\/|\\]titlepage\./ =~ e }
-end
+newtitlepage = checkForSubmittedTitlepages(submitted_images, 'check_for_submitted_titlepages')
 
 # if an epub-specific titlepage file has been submitted, use that;
 # otherwise use the new POD coverpage if it exists;
 # and if neither exists, we'll create the epubtitlepage, and set it to the final archival path.
 # (POD titlepage images should only be submitted manually by the user, never created programatically.)
 
-final_cover = oldtitlepage
-
-if File.file?(newtitlepage)
-  final_cover = newtitlepage
-end
-
-# set the default switch to generate the titlepage
-gen = false
+final_cover = setFinalCover(oldtitlepage, newtitlepage, 'set_final_cover_var')
 
 # Determine whether or not to generate a titlepage.
-# First check: if an epub titlepage has previously been generated, and no new image has been submitted by the user
-if File.file?(titlepagelog) and !File.file?(newtitlepage)
-  gen = true
-  Mcmlln::Tools.deleteFile(titlepagelog)
-  Mcmlln::Tools.deleteFile(oldtitlepage)
-# Then check: if a titlepage has previously been generated, but there IS a new image submitted by the user
-elsif File.file?(titlepagelog) and File.file?(newtitlepage)
-  gen = false
-  Mcmlln::Tools.deleteFile(titlepagelog)
-  Mcmlln::Tools.deleteFile(oldtitlepage)
-# Finally: if no titlepage has ever been generated, and no new image has been submitted,
-# and there is no existing image archived from a previous run
-elsif !File.file?(titlepagelog) and !File.file?(final_cover)
-  gen = true
-end
-
+gen = decideTitlepageGeneration(titlepagelog, newtitlepage, oldtitlepage, final_cover, 'determine_titlepage_generation')
+@log_hash['gen(erate_titlepage)_value'] = gen
 puts "Titlepage generation: #{gen}"
 
 # now that we've got the logic out of the way,
-# set a default value for the final titlepage, 
+# set a default value for the final titlepage,
 # if no images were found above
 
 if final_cover.empty? or final_cover.nil?
   final_cover = arch_epubtp
 end
 
 # CSS that will format the final titlepage PDF
-if File.file?("#{Bkmkr::Paths.scripts_dir}/covermaker/css/#{project_dir}/titlepage.css")
-  cover_css_file = File.join(pdf_css_dir, project_dir, "titlepage.css")
-else
-  cover_css_file = File.join(pdf_css_dir, "generic", "titlepage.css")
-end
+#this method was checking the css path based project_dir from metadata, which at this stage _
+#in our deploy.bad only holds 'N/A'.  So commenting out the chooseCSS method and it's call here:
+# cover_css_file = chooseCss(project_dir, pdf_css_dir, 'choose_cover_css_file')
```

Comment:
Also, this answers my question from above. So, as long as you test and everything seems to be working, you can delete the old method.